### PR TITLE
feat: Added support for GitHub Enterprise instances (#10).

### LINKED ISSUE #10: Add support for Github enterprise
### TL;DR

Support scanning GHES for misconfigurations

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by SvenTo**
This pull request adds support for GitHub Enterprise instance (#10).

#### What's being changed?

You can run legitify against a GitHub Enterprise instance if you set the endpoint URL in the environment variable ``GITHUB_ENDPOINT``:

```sh
export GITHUB_ENDPOINT="https://github.example.com/"
GITHUB_TOKEN=<your_token> legitify analyze --org org1,org2 --namespaces organization,member
```

#### Is this PR related to an existing issue?

resolves #10 

#### Check off the following:

- [x] This PR follows the CONTRIBUTION.md guidelines
- [x] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ GITHUB_TOKEN=<your_token> legitify analyze --org org1,org2 --namespaces organiza
 ```
 The above command will test organization and member policies against org1 and org2.
 
+## GitHub Enterprise Support
+You can run legitify against a GitHub Enterprise instance if you set the endpoint URL in the environment variable ``GITHUB_ENDPOINT``:
+
+```sh
+export GITHUB_ENDPOINT="https://github.tools.sap/"
```
Comment (gal-legit) [inline README.md:57]:
```suggestion
export GITHUB_ENDPOINT="https://private.github.instance/"
```

Comment (gal-legit) [inline README.md:57]:
```suggestion
export GITHUB_ENDPOINT="https://github.example.com/"
```

File: internal/clients/github/client.go [hunk 2]
```diff
@@ -220,6 +244,11 @@ func (c *client) getRole(orgName string) (permissions.OrganizationRole, error) {
 
 func (c *client) collectTokenScopes() (permissions.TokenScopes, error) {
 	graphQLUrl := "https://api.github.com/graphql"
+	githubEndpoint := viper.GetString(common_options.EnvGitHubEndpoint)
+	if githubEndpoint != "" {
+		graphQLUrl = getGitHubEnterpriseGraphURL()
+	}
+
```
Comment (gal-legit) [inline internal/clients/github/client.go:251]:
```suggestion
	graphQLUrl := getGitHubGraphURL()
```

**Comment by gal-legit** [comment]
Hi @SvenTo, thank you for your valuable contribution to Legitify!
I made a few suggestions, please feel free to comment on these.
We're reviewing and running the tests and will approve the PR shortly.

File: internal/clients/github/client.go [hunk 3]
```diff
@@ -68,11 +80,27 @@ func NewClient(ctx context.Context, token string, org []string, fillCache bool)
...
 
 	acceptHeader := experimentalApiAcceptHeader
 	clientWithAcceptHeader := NewClientWithAcceptHeader(tc.Transport, &acceptHeader)
-	graphQLClient := githubv4.NewClient(&clientWithAcceptHeader)
+
+	var graphQLClient *githubv4.Client
+	if githubEndpoint == "" {
+		graphQLClient = githubv4.NewClient(&clientWithAcceptHeader)
+	} else {
+		graphQLClient = githubv4.NewEnterpriseClient(getGitHubEnterpriseGraphURL(), &clientWithAcceptHeader)
```
Comment (gal-legit) [inline internal/clients/github/client.go:102]:
```suggestion
		graphQLClient = githubv4.NewEnterpriseClient(getGitHubGraphURL(), &clientWithAcceptHeader)
```

**Comment by SvenTo** [comment]
@gal-legit Thanks for the feedback. I've resolved all suggestions.

**Review by gal-legit — APPROVED** [review]

**Comment by gal-legit** [comment]
@SvenTo perfect! thanks again for your contribution and cooperation :)

**Pull request merged**